lib/screens/home/home_view_model.dart

```diff
@@ -69,26 +85,55 @@ class HomeViewModel extends AutoDisposeAsyncNotifier<void> {
   }
 
   Future<void> getSurveyList() async {
+    _pageNumber += 1;
+    if (_numberOfPage != null && _pageNumber > (_numberOfPage ?? 0)) {
+      return;
+    }
+    _getSurveysFromNetwork();
+
     if (_surveyCache.isEmpty) {
       showOrHideShimmer(true);
     } else {
       showOrHideLoadMore(true);
     }
+  }
 
-    // TODO replace with usecase in integration
-    Future.delayed(const Duration(seconds: 2), () {
-      _surveyCache.addAll((_mockList(_surveyCache.length) as Success).value);
+  void _getSurveysFromNetwork() async {
+    final result = await ref.read(getSurveysUseCaseProvider).call(
+          SurveysParameters(
+            pageNumber: _pageNumber,
+            pageSize: pageSize,
+          ),
+        );
+
+    if (result is Success) {
+      final successValue = (result as Success<SurveyDataResponse>).value;
+      _surveyCache.addAll(successValue.surveys
+          .map((survey) => survey.toSurveyModel())
+          .toList());
+      _numberOfPage = successValue.meta.pages;
       _surveys.add(_surveyCache);
-      showOrHideShimmer(false);
-      showOrHideLoadMore(false);
-    });
+    } else if (result is Failed) {
+      _getCacheSurveys();
+      _isError.add((result as Failed).getErrorMessage());
+    }
+    showOrHideShimmer(false);
+    showOrHideLoadMore(false);
```

Comment:
Then we can remove these methods
```suggestion
    } else if (result is Failed) {
      _getCacheSurveys();
      _isError.add((result as Failed).getErrorMessage());
    }
```